js/wp-seo-metabox-category.js

```diff
@@ -0,0 +1,217 @@
+/* global wp, _, wpseoPrimaryCategoryL10n */
+(function( $ ) {
+	'use strict';
+
+	var primaryTermInputTemplate, primaryTermUITemplate, primaryTermScreenReaderTemplate;
+	var taxonomies = wpseoPrimaryCategoryL10n.taxonomies;
+
+	/**
+	 * Checks if the elements to make a term the primary term and the display for a primary term exist
+	 *
+	 * @param checkbox
+	 *
+	 * @return {bool}
+	 */
+	function hasPrimaryTermElements( checkbox ) {
+		return 1 === $( checkbox ).closest( 'li' ).children( '.wpseo-make-primary-term' ).length;
+	}
+
+	/**
+	 * Retrieves the primary term for a taxonomy
+	 *
+	 * @param taxonomyName
+	 * @returns {string}
+	 */
+	function getPrimaryTerm( taxonomyName ) {
+		var primaryTermInput;
+
+		primaryTermInput = $( '#yoast-wpseo-primary-' + taxonomyName );
+		return primaryTermInput.val();
+	}
+
+	/**
+	 * Sets the primary term for a taxonomy
+	 *
+	 * @param taxonomyName
+	 * @param termId
+	 */
+	function setPrimaryTerm( taxonomyName, termId ) {
+		var primaryTermInput;
+
+		primaryTermInput = $( '#yoast-wpseo-primary-' + taxonomyName );
+
+		primaryTermInput.val( termId );
+	}
+
+	/**
+	 * Makes the first term primary for a certain taxonomy
+	 *
+	 * @param taxonomyName
+	 */
+	function makeFirstTermPrimary( taxonomyName ) {
+		var firstTerm = $( '#' + taxonomyName + 'checklist input[type="checkbox"]:checked:first' );
+
+		setPrimaryTerm( taxonomyName, firstTerm.val() );
+		updatePrimaryTermSelectors( taxonomyName );
+	}
+
+	/**
+	 * Updates the primary term selectors/indicators for a certain taxonomy
+	 *
+	 * @param taxonomyName
+	 */
+	function updatePrimaryTermSelectors( taxonomyName ) {
+		var checkedTerms, uncheckedTerms;
+		var listItem, label;
+
+		checkedTerms = $( '#' + taxonomyName + 'checklist input[type="checkbox"]:checked' );
```

Comment:
Could you assign `$( '#' + taxonomyName + 'checklist input[type="checkbox"])` to a variable, and then `filter()` with `:checked` / `:not(:checked)` as needed? That would save having to find all of the checkboxes inside an ID'd element twice.